src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -476,14 +491,14 @@ Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClus
 
 Instance::~Instance()
 {
-    ModeBaseAliasesInstances.erase(this);
+    UnregisterInstance(this);
     chip::app::InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
     unregisterAttributeAccessOverride(this);
 }
 
-std::set<Instance *> * GetModeBaseInstances()
+IntrusiveList<Instance> & GetModeBaseInstanceList()
```

Comment:
Should this be returning `const IntrusiveList<Instance> &`?  Do we expect consumers to modify the list?  I'd guess no.

Reply:
I did try this but for some reason, when the OnOff server tries to do a for loop on the contents of the `IntrusiveList`, the elements it returns are also const and therefore it cannot use the `UpdateXxx` functions, which is required.

Reply:
Please document that.

Containers and const on a container means all the things in it are also const are annoying...

Reply:
I added this comment above the getter `// This does not return a const of gModeBaseAliasesInstances since the caller might need to call the element's UpdateXxx methods.`.